src/analyze-dependencies.ts

```diff
@@ -0,0 +1,273 @@
+ * 
+ * // Enable all debug logs
+ * logger.setOptions({ enabled: true, level: 'debug' });
+ * 
+ * // Or create a specific logger for dependency analysis
+ * const analyzerLogger = logger.child('analyzer');
+ * analyzerLogger.setOptions({ enabled: true, level: 'debug' });
+ * ```
+ */
+
+// Re-export types
+export type { DependencyStats, DependencyAnalyzer };
+
+export class LocalDependencyAnalyzer implements DependencyAnalyzer {
+  async analyzeDependencies(root: string): Promise<DependencyStats> {
+    try {
+      const pkgJsonPath = path.join(root, 'package.json');
+      logger.debug('Reading package.json from:', pkgJsonPath);
+
+      const pkgJson = JSON.parse(await fs.readFile(pkgJsonPath, 'utf-8'));
+
+      // Count direct dependencies
+      const directDependencies = Object.keys(pkgJson.dependencies || {}).length;
+      const devDependencies = Object.keys(pkgJson.devDependencies || {}).length;
+
+      logger.debug('Direct dependencies:', directDependencies);
+      logger.debug('Dev dependencies:', devDependencies);
+
+      // Analyze node_modules
+      let cjsDependencies = 0;
+      let esmDependencies = 0;
+      let installSize = 0;
+
+      // Walk through node_modules
+      const nodeModulesPath = path.join(root, 'node_modules');
+
+      try {
+        await fs.access(nodeModulesPath);
+        logger.debug('Found node_modules directory');
+
+        await this.walkNodeModules(nodeModulesPath, {
+          onPackage: (pkgJson) => {
+            const type = analyzePackageModuleType(pkgJson);
+            logger.debug(`Package ${pkgJson.name}: ${type} (type=${pkgJson.type}, main=${pkgJson.main}, exports=${JSON.stringify(pkgJson.exports)})`);
+
+            if (type === 'cjs') cjsDependencies++;
+            if (type === 'esm') esmDependencies++;
+            if (type === 'dual') {
+              cjsDependencies++;
+              esmDependencies++;
+            }
+          },
+          onFile: (filePath) => {
+            try {
+              const stats = fsSync.statSync(filePath);
+              installSize += stats.size;
+            } catch {
+              logger.debug('Error getting file stats for:', filePath);
+            }
+          }
+        });
+      } catch {
+        logger.debug('No node_modules directory found');
+      }
+
+      logger.debug('Analysis complete:');
+      logger.debug('- CJS dependencies:', cjsDependencies);
+      logger.debug('- ESM dependencies:', esmDependencies);
+      logger.debug('- Install size:', installSize, 'bytes');
+
+      return {
+        totalDependencies: directDependencies + devDependencies,
+        directDependencies,
+        devDependencies,
+        cjsDependencies,
+        esmDependencies,
+        installSize,
+        packageName: pkgJson.name,
+        version: pkgJson.version
+      };
+    } catch (error) {
+      logger.error('Error analyzing dependencies:', error);
+      throw error;
+    }
+  }
+
+  private async walkNodeModules(
```

Comment:
we can probably simplify this massively by using something like `fdir`, maybe for a follow up PR

Reply:
should we open an issue for this to keep track?

Reply:
that'd be a good idea I think 👍